aea/skills/base.py

```diff
@@ -55,7 +55,7 @@ def agent_name(self) -> str:
         return self._agent_context.agent_name
 
     @property
-    def agent_public_key(self) -> str:
+    def agent_public_key(self) -> Dict[str, str]:
```

Comment:
Use this for the `default` public key so the interface is stable but add the whole dictionary as `agent_public_keys`.